test/modules/TestAnalyzer/ModuleMessagesReport.cs

```diff
@@ -42,7 +42,7 @@ public ModuleMessagesReport(string moduleId, StatusCode statusCode, long receive
 
         public IList<MissedMessagesDetails> MissedMessages { get; }
 
-        public bool IsPassed => this.StatusCode == StatusCode.AllMessages;
+        public bool IsPassed => this.StatusCode == StatusCode.AllMessages || this.StatusCode == StatusCode.OldMessages;
```

Comment:
Can we make a new status code, something like OldMessagesNoneMissing and OldMessagesWithMissing?

This will currently pass even with missing messages as long as they are also old messages. I think we should fail if there are missing messages, but succeed on Old messages as long as there are none missing

Reply:
Oh wow great catch! I thought it checked for old messages after missing. Do we need a new status code, or can we just re-order the checks?

Reply:
Yep, ordering works even better

Reply:
I'm glad we caught this before 1.0.10. This has the potential to completely mask message loss. Luckily I'm pretty confident about our prior releases as arm32 never showed message loss.

Reply:
@dylanbronson Once I started refactoring I liked your idea of a custom status code for missing and old messages together. I implemented it this way.

Reply:
Do we need status code?  If doing it again, I will store last result timestamp (maybe also threshold) and some counts.

Reply:
If we didn't have a status code we would need similar logic in order to determine IsPassed for the report. I like the status code because it provides more information than just true / false.

Reply:
In TestResultReporter https://github.com/Azure/iotedge/blob/a5d4eff8da25ab37b931fdd17ca1c79d419e7f44/test/modules/TestAnalyzer/TestResultReporter.cs#L105, it returns statusCode.OldMessages even missingCounter > 0.  This means we will pass the test even there is missing message.  Am I missing anything?

Reply:
I'm confused. What you linked to is the current implementation in master. You are correct that this is not what we want, so we reordered the checks in this PR.

Reply:
Sorry, I forget to refresh PR page and shows the old version.  You are good and comments are provided.